File: Next_Permutation/submission3.py
class Solution(object):
    def getSmallestIndex(self, nums, ptr, n):

        smallest = ptr + 1
        for i in range(ptr+2, n):
            if nums[i] > nums[ptr] and nums[i] < nums[smallest]:
                smallest = i
        return smallest

    def nextPermutation(self, nums):
        
        ptr = -1
        n = len(nums)

        for i in range(n-1):
            if (nums[i]<nums[i+1]):
                ptr = i
        
        if ptr==-1:
            return nums.sort()
        else:
            index = self.getSmallestIndex(nums, ptr, n)
            print(index)
            nums[ptr], nums[index] = nums[index], nums[ptr]
            nums[ptr+1:] = sorted(nums[ptr+1:])
            return nums

File: Longest_Consecutive_Sequence/submission3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size(), ans = 0, current = 0;
        sort(nums.begin(), nums.end());

        for (int i=0; i<n; i++){
            if (i==0 || nums[i]-nums[i-1]==1) current++;
            else if (nums[i]-nums[i-1]!=0) current = 1;
            ans = max(ans, current);
        }

        return ans;
    }
};